pkg/build/util/secrets.go

```diff
@@ -0,0 +1,60 @@
+package util
+
+import (
+	"encoding/json"
+	"strings"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/credentialprovider"
+	"github.com/golang/glog"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+// DockerConfigFromSecret parse the Secret 'data' and return the DockerConfig
+// which contains representation of the '.dockercfg'.
+// FIXME: This was copied from upstream because the upstream function was not
+//        public.
+func DockerConfigFromSecret(secret *kapi.Secret) (cfg *credentialprovider.DockerConfig) {
+	if secret.Type != kapi.SecretTypeDockercfg {
+		return nil
+	}
+	if err := json.Unmarshal(secret.Data[kapi.DockerConfigKey], cfg); err != nil {
+		glog.V(2).Infof("Unable to parse the .dockercfg content in secret %s: %v", secret.Name, err)
+		return nil
+	}
+	return cfg
+}
+
+// HasDockerConfigEntry checks if the Docker image (represented as full Docker
+// pull spec format) has entry for its registry defined in the provided
+// DockerConfig
+func HasDockerConfigEntry(imageSpec string, cfg credentialprovider.DockerConfig) bool {
+	image, err := imageapi.ParseDockerImageReference(imageSpec)
+	if err != nil {
+		glog.V(2).Infof("Unable to parse the Docker image '%s': %v", imageSpec, err)
+		return false
+	}
+	registry := image.DockerClientDefaults().Registry
```

Comment:
@ncdc do you know if we already have code in OpenShift/Kube that handles this 'docker.io' vs "https://index.docker.io/v1" I can reuse? I don't like making this hairy :-)